http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -783,7 +791,7 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       }
 
       // create second Galaxy disk asynchronously
-      createSecondDiskOp = if (msg.appType == Galaxy) {
+      createSecondDiskOp = if (msg.appType == Galaxy && msg.createDisk.isDefined) {
```

Comment:
Do we need to check `msg.createDisk.isDefined` above (on line 782) as well?

Reply:
nope..`createDiskForApp` function itself is doing the check already..this is why I prefer having the important logic in for comprehension instead of having it in small functions like this, kinda makes things confusing (having to look at the function to realize it's already doing the check)

Reply:
Where in the `createDiskForApp` function is that check being done again?

Reply:
per my previous comment, I refactored `createDiskForApp`. The original function looks like [this](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala#L1087), which is already doing the check.

Reply:
oh okay, thanks…I was afraid I was losing my mind 🤯

Reply:
ok - yeah looks like `msg.createDisk` is being checked in both places.